spring-web/src/test/java/org/springframework/web/util/ContentCachingRequestWrapperTests.java

```diff
@@ -117,4 +119,40 @@ void inputStreamFormPostRequest() throws Exception {
 		assertThat(wrapper.getContentAsByteArray()).isEqualTo(response);
 	}
 
+	/**
+	 * When content limiting, if the content length is known and is lower than the content limit, the internal
+	 * buffer should allocate that size. This exaggerates this scenario by using a very large content limit that
+	 * would cause an OOM Error likely otherwise.
+	 */
+	@Test
+	void cachedContentWithLimitKnownLengthAvoidOverAllocation() throws Exception {
```

Comment:
This is a bit awkward to test without exposing some internals in a package private method or something. Basically, using a really large content limit before this change could cause an OOM Error if you ran this test with < 2GB heap or so. After this change it wouldn't use the 2GB limit, it would just allow growing to it (but the tests don't exercise growing to it as that's not relevant here).